app/controllers/inventories_controller.rb

```diff
@@ -0,0 +1,77 @@
+class InventoriesController < ApplicationController
+  load_and_authorize_resource
+  before_action :set_inventory, only: %i[show destroy]
+
+  # GET /inventories
+  def index
+    @inventories = current_user.inventories
+  end
+
+  # GET /inventories/1
+  def show
+    @inventory = Inventory.includes(:inventory_foods).find(params[:id])
+  end
+
+  def shopping_list
+    @inventory = Inventory.includes(:inventory_foods).find(params[:inventory_id])
+    @recipe = Recipe.includes(:recipe_foods).find(params[:recipe_id])
+
+    @items = []
+    @total_amount = 0
+    @recipe.recipe_foods.find_each do |recipe_food|
+      inventory_food = @inventory.inventory_foods.find_by(food: recipe_food.food)
+      difference = inventory_food.quantity - recipe_food.quantity
```

Comment:
- [ ] When I try to generate a shopping list, the application crashes. if I don't have a food item in my inventory. 

![Screenshot 2021-11-19 at 00-45-10 Action Controller Exception caught](https://user-images.githubusercontent.com/1642014/142502460-48d524ae-68d0-4af7-b293-a3c8694edf3d.png)